src/components/PeopleTable.tsx

```diff
@@ -1,5 +1,103 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-export const PeopleTable = () => {
+import React, { useState } from 'react';
+import { Person } from '../types';
+import { SearchLink } from './SearchLink';
+import classNames from 'classnames';
+import { useLocation, useNavigate } from 'react-router-dom';
+
+type PeopleTableProps = {
+  people: Person[];
+  selectedSlug: string;
+};
+
+export const PeopleTable: React.FC<PeopleTableProps> = ({
+  people,
+  selectedSlug,
+}) => {
+  const location = useLocation();
+  const navigate = useNavigate();
+
+  const [sortField, setSortField] = useState<string | null>(null);
+  const [sortOrder, setSortOrder] = useState<'asc' | 'desc' | null>(null);
+
+  const queryParams = new URLSearchParams(location.search);
+  const initialSortField = queryParams.get('sort');
+  const initialSortOrder = queryParams.get('order') as 'asc' | 'desc' | null;
+
+  React.useEffect(() => {
+    if (initialSortField) {
+      setSortField(initialSortField);
+      setSortOrder(initialSortOrder);
+    }
+  }, [initialSortField, initialSortOrder]);
+
+  const handleSort = (field: string) => {
+    if (field === sortField) {
+      setSortOrder(prevOrder => (prevOrder === 'asc' ? 'desc' : 'asc'));
+    } else {
+      setSortField(field);
+      setSortOrder('asc');
+    }
+  };
+
+  React.useEffect(() => {
+    if (sortField) {
+      const searchParams = new URLSearchParams();
+
+      searchParams.set('sort', sortField);
+      if (sortOrder) {
+        searchParams.set('order', sortOrder);
+      }
+
+      navigate({ search: searchParams.toString() });
+    } else {
+      navigate({ search: '' });
+    }
+  }, [sortField, sortOrder, navigate]);
+
+  const sortedPeople = [...people];
+
+  if (sortField && sortOrder) {
+    sortedPeople.sort((a, b) => {
+      const aValue = a[sortField as keyof Person];
+      const bValue = b[sortField as keyof Person];
+
+      if (
+        aValue === null ||
+        aValue === undefined ||
+        bValue === null ||
+        bValue === undefined
+      ) {
+        return 0;
```

Comment:
The sorting logic currently returns 0 if either `aValue` or `bValue` is null or undefined. This could lead to unexpected sorting behavior. Consider handling these cases more explicitly, perhaps by placing null or undefined values at the end of the sorted list.